polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -686,6 +617,110 @@ impl<T: Config> Pallet<T> {
 		};
 		Ok((processed, Some(all_weight_after).into()))
 	}
+
+	fn back_candidates(
+		context: ProcessInherentDataContext,
+		concluded_invalid_hashes: BTreeSet<CandidateHash>,
+		backed_candidates: Vec<BackedCandidate<T::Hash>>,
+	) -> Result<
+		(
+			Vec<(CandidateReceipt<T::Hash>, Vec<(ValidatorIndex, ValidityAttestation)>)>,
+			BTreeMap<ParaId, Vec<(BackedCandidate<T::Hash>, CoreIndex)>>,
+		),
+		DispatchErrorWithPostInfo,
+	> {
+		let allowed_relay_parents = shared::AllowedRelayParents::<T>::get();
+		let upcoming_new_session = initializer::Pallet::<T>::upcoming_session_change();
+
+		METRICS.on_candidates_processed_total(backed_candidates.len() as u64);
+
+		let initial_candidate_count = backed_candidates.len();
+
+		if !upcoming_new_session {
```

Comment:
Let's also modify collators to skip block production here. This way we can make the service smoother for them (no dropped blocks).